lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -6559,19 +7074,52 @@ impl FunctionCodeGenerator<CodegenError> for X64FunctionCode {
 
                 if !was_unreachable && frame.returns.len() > 0 {
                     let loc = *self.value_stack.last().unwrap();
-                    Self::emit_relaxed_binop(
-                        a,
-                        &mut self.machine,
-                        Assembler::emit_mov,
-                        Size::S64,
-                        loc,
-                        Location::GPR(GPR::RAX),
-                    );
+                    match frame.returns[0] {
+                        WpType::F32 | WpType::F64 => {
```

Comment:
I think this is correct because you can apply canonicalization early, but I think this is unnecessary?

Reply:
Code reference here is outdated so I'm not sure which part you are referring to; but if you mean the `Phi`-like branch like `Br`-ing out, early canonicalization is necessary because we don't have the static tracking information after.